User/InMemoryUserRepository.php

```diff
@@ -92,22 +92,22 @@ public function getSupportedClass()
     /**
      * {@inheritdoc}
      */
-    public function add(UserInterface $user)
+    public function add(User $user)
     {
         $this->users[] = $user;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function save(UserInterface $user)
+    public function save(User $user)
```

Comment:
save is the method used when updating but this happens automagically when editing the original object (at least in memory)